lib/postprocessing/image.js

```diff
@@ -16,10 +16,163 @@ const gm = require("./gm-promisify");
 const util = require('util');
 const exec = util.promisify(require('child_process').exec);
 
+// list of supported output formats
+// maps our fmt ('png') to imagemagick format ('PNG') https://imagemagick.org/script/formats.php
+const SUPPORTED_OUTPUT_FMT = {
+    "png": "PNG",
+    "jpg": "JPEG",
+    "jpeg": "JPEG",
+    "tif": "TIFF",
+    "tiff": "TIFF",
+    "gif": "GIF"
+};
+
+// list of supported input formats (intermediate rendition)
+// in imagemagick format https://imagemagick.org/script/formats.php
+const SUPPORTED_INPUT_FMT = new Set([
+    "PNG",
+    "JPEG",
+    "JPEG",
+    "TIFF",
+    "TIFF",
+    "GIF",
+    "BMP"
+]);
+
+// minimal instructions which many workers can already handle themselves
+// const BASIC_INSTRUCTIONS = new Set([
+//     "fmt",
+//     "width",
+//     "height",
+//     "quality"
+// ]);
+
+// list of post processing instructions we handle
+const EXTRA_INSTRUCTIONS = new Set([
+    "crop",
+    "jpegSize",
+    "interlace",
+    "dpi",
+    "convertToDpi"
+]);
+
+function isSupportedOutputFormat(instructions) {
+    return Object.keys(SUPPORTED_OUTPUT_FMT).includes(instructions.fmt);
+}
+
+function isSupportedInputFormat(metadata) {
+    return SUPPORTED_INPUT_FMT.has(metadata.format);
+}
+
+function hasExtraInstructions(instructions) {
+    return Object.keys(instructions).some(key => EXTRA_INSTRUCTIONS.has(key));
+}
+
+async function getImageMetadata(file) {
+    try {
+        // [1x1+0+0] is a trick to avoid reading all pixels into memory, improving time for large images
+        const result = await gm(`${file}[1x1+0+0]`).write("json:-");
```

Comment:
we do this command also now in worker-flite. would be nice to store this information somewhere (maybe in a nice JSON with only the important information, maybe on the rendition) so we can re-use this inside worker-flite and won't have to call the command twice (although the command is pretty fast)

Reply:
for reference: https://git.corp.adobe.com/nui/worker-flite/blob/aa8b2a841a8b46ba1ac7d726e169a1b2173fa72d/action/worker.js#L143-L158

Reply:
nevermind, both are needed. This is examining the rendition metadata...in flite we examine the source metadata

Reply:
They run on different files: Here it runs on the intermediate rendition, whereas in worker-flite you inspect the source.

Regarding a reusable API: I think since flite is the only other case right now, "duplicating" that code once is fine. Having to add this API, support it and create some coupling here (what if on source files we need to slightly tweak the command?) seems like extra cost/risk that we might want to avoid for now.

Definitely worth thinking about if the need comes up in other places in the future.